src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -419,112 +423,90 @@ def _reject_all_pending_requests(self):
 
     def set_energy_forecast(self, payload: Dict) -> None:
         """Callback for set_energy_forecast redis endpoint."""
+        self.set_device_energy_data(payload, "set_energy_forecast", "energy_forecast")
+
+    def set_energy_measurement(self, payload: Dict) -> None:
+        """Callback for set_energy_measurement redis endpoint."""
+        self.set_device_energy_data(payload, "set_energy_measurement", "energy_measurement")
+
+    def set_device_energy_data(self, payload: Dict, command_type: str,
```

Comment:
I like this, but this will make the unittests fail (that I recently added).
I dont know how we should continue here. Do you want to adapt to my branch once again or wait until I pushed my changes to master

Reply:
Hmmm CI did not complain, that's why I was rest assured that my changes did not break the test. Seems like this is not the case. 

At any case don't worry about it, I'll fix the tests too. Thanks for pinging me